### PR TITLE
External networks belong to datacenter

### PR DESCRIPTION
**Pull request by gekorob**
In ovirt external networks with same name can be defined in the same ems but
not in the same datacenter.
We need a way to let external_distributed_virtual_switches belong to the
owning datacenter

Related to:
  - https://bugzilla.redhat.com/show_bug.cgi?id=1751151
  - https://github.com/ManageIQ/manageiq-providers-ovirt/pull/477
  - https://github.com/ManageIQ/manageiq-providers-ovirt/pull/478

### CODE REVIEW

**Comment by gekorob** [comment]
@agrare, need a way to add *external_distributed_virtual_switches* to *dependency_collections* that are used in the core
https://github.com/ManageIQ/manageiq/blob/6ccb6c546289dcd67d25415e4b6de943566cfbb3/app/models/manageiq/providers/inventory/persister/builder/infra_manager.rb#L288

**Comment by gekorob** [comment]
Kudos to @agrare for https://github.com/ManageIQ/manageiq-providers-ovirt/pull/480/commits/331cde027b0efe8714e93f00580c6ebfd726d886

File: app/models/manageiq/providers/redhat/inventory/parser/infra_manager.rb [hunk 1]
```diff
@@ -17,12 +17,24 @@ def parse
...
+        :name    => network.name,
+        :uid_ems => network.id
+      }
+
+      if is_external
+        datacenter = network.data_center
+        ems_ref = ManageIQ::Providers::Redhat::InfraManager.make_ems_ref(datacenter.href)
+        parent_datacenter = persister.datacenters.lazy_find(ems_ref)
+
+        attrs_to_assign[:parent] = parent_datacenter if id_of_external_network?(network.id)
```
Comment (agrare) [inline app/models/manageiq/providers/redhat/inventory/parser/infra_manager.rb]: Minor cleanup, if we are in `is_external` then `if id_of_external_network?(network.id)` will always be true

File: app/models/manageiq/providers/redhat/inventory/persister/infra_manager.rb [hunk 2]
```diff
@@ -52,6 +45,13 @@ def initialize_inventory_collections
         :complete                     => !targeted?
       )
     end
+    add_collection(infra, :parent_blue_folders)
```
Comment (agrare) [inline app/models/manageiq/providers/redhat/inventory/persister/infra_manager.rb]: Another minor cleanup item, you don't need to add the collection here and also [here](https://github.com/ManageIQ/manageiq-providers-ovirt/pull/480/files#diff-85616c128d7cc811ee4dc4c3225588d9R114)

File: app/models/manageiq/providers/redhat/inventory/parser/infra_manager.rb (hunk 1 above)
Comment (gekorob) [reply app/models/manageiq/providers/redhat/inventory/parser/infra_manager.rb]: Oh yes, sorry I left it behind... I'll fix it in a new cleanup review

File: app/models/manageiq/providers/redhat/inventory/persister/infra_manager.rb (hunk 2 above)
Comment (gekorob) [reply app/models/manageiq/providers/redhat/inventory/persister/infra_manager.rb]: Absolutely, I'll fix it in the cleanup PR

**Comment by gekorob** [comment]
@agrare really thx for the help on this PR

**Comment by simaishi** [comment]
Adding `jansa/yes` as https://github.com/ManageIQ/manageiq-providers-ovirt/pull/478 is `jansa/yes` and depends on this PR.

**Comment by simaishi** [comment]
Jansa backport details:

```
$ git log -1
commit 7bf5f77ac2a9a515e37c12acc6fb9d7b11ee8d13
Author: Adam Grare <agrare@redhat.com>
Date:   Wed Mar 4 09:57:59 2020 -0500

    Merge pull request #480 from gekorob/ext_nets_as_folders

    External networks belong to datacenter

    (cherry picked from commit 27fffa642d37a506ddca91fdef7af8846f483215)

    https://bugzilla.redhat.com/show_bug.cgi?id=1751151
```

**Pull request merged**